packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -608,10 +608,63 @@ module.exports = (
             let timeToRead = 0
             const pureText = sanitizeHTML(html, { allowTags: [] })
             const avgWPM = 265
+
+            // Unicode ranges for Han (Chinese) and Hiragana/Katakana (Japanese) characters
+            const cjRanges = [
+              [11904, 11930], // Han
+              [11931, 12020],
+              [12032, 12246],
+              [12293, 12294],
+              [12295, 12296],
+              [12321, 12330],
+              [12344, 12348],
+              [13312, 19894],
+              [19968, 40939],
+              [63744, 64110],
+              [64112, 64218],
+              [131072, 173783],
+              [173824, 177973],
+              [177984, 178206],
+              [178208, 183970],
+              [183984, 191457],
+              [194560, 195102],
+              [12353, 12439], // Hiragana
+              [12445, 12448],
+              [110593, 110879],
+              [127488, 127489],
+              [12449, 12539], // Katakana
+              [12541, 12544],
+              [12784, 12800],
+              [13008, 13055],
+              [13056, 13144],
+              [65382, 65392],
+              [65393, 65438],
+              [110592, 110593],
+            ]
+
+            function isCjChar(char) {
+              const charCode = char.codePointAt(0)
+              return cjRanges.some(
+                ([from, to]) => charCode >= from && charCode < to
+              )
+            }
+
+            let latinChars = []
+            let cjChars = []
+
+            for (const char of pureText) {
+              if (isCjChar(char)) {
+                cjChars.push(char)
+              } else {
+                latinChars.push(char)
+              }
+            }
```

Comment:
This become quite big function with a lot of nuance. I feel like with extra complexity it deserves to be moved to it's own file - maybe `packages/gatsby-transformer-remark/src/utils/time-to-read.js` (?)

Just few notes - it would be good for `cjRanges` array and `isCjChar` function to be defined outside of "timeToRead" function to not init those for each call of that function - they are immutable array and pure function so it's perfectly fine to reuse same instances of those and not re-init those.

Reply:
Done in 9cebafada1ef82afb2ab30812ce84baa728e7549 !

I also think that a larger refactor is needed, because `wordCount` and `timeToRead` do their own counting which is not very good. But that is another issue - I created one in #21357

Reply:
> @pieh Done in 9cebafa !

Awesome!

>  also think that a larger refactor is needed, because wordCount and timeToRead do their own counting which is not very good. But that is another issue - I created one in #21357

Good point and also good intuition to do this separately. My comment here was resolved, so I'll just "resolve" that thread and do final review